logger/context.go

```diff
@@ -0,0 +1,93 @@
+package logger
+
+import (
+	"bytes"
+	"encoding"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+var (
+	_ encoding.TextMarshaler = LogContext{}
+)
+
+type LogUser interface {
+	GetID() uint
+	GetEmail() string
+}
+
+type LogContext struct {
+	Data    map[string]interface{}
+	Error   error
+	Request *http.Request
+	User    LogUser
+}
+
+// MarshalText converts LogContext into a JSON representation, eliminating zero-value fields.
+//
+// Values in LogContext.Data that cannot be represented in JSON will cause an error to be thrown.
+//
+// MarshalText implements encoding.TextMarshaler.
```

Comment:
Maybe you can help me understand why this is necessary?

Reply:
Why we need a marshaler or why I used `encoding.TextMarshaler`?

Reply:
Yah, not having much context, just a noob question. Trying to understand why this was needed.

Reply:
Not a newb question at all - the primary reason is I wanted to include the `*http.Request` in the log statement, but that struct has functions that cannot be serialized into JSON. So, manually marshal it so I can skirt around those. Since I'm already doing that, I go ahead and hand-write the `map[string]interface{}` that is eventually passed to `json.Marshal` to be super explicit about the schema of this object.